lang/java/reef-utils/src/main/java/org/apache/reef/util/Optional.java

```diff
@@ -35,19 +35,15 @@
   private static final long serialVersionUID = 42L;
 
   private final T value;
-  private final String valueStr;
-  private final int valueHash;
+  private static final String EMPTY_VALUE_STR = "OptionalvNothing";
+  private static final int EMPTY_VALUE_HASH = 0;
```

Comment:
Maybe choose something different than `0`, which might trigger edge cases in other code. A prime number is commonly chosen.